torch/_dynamo/decorators.py

```diff
@@ -58,15 +72,35 @@ def disable(fn=None, recursive=True):
     but still process recursively invoked frames.
     """
     if recursive:
+        ctx = (
+            OldDisableContext()
+            if _old_compile_disable_behavior()
+            else CompileEnabledContext(False)
+        )
         if fn is not None:
-            fn = innermost_fn(fn)
+            if _old_compile_disable_behavior():
+                fn = innermost_fn(fn)
```

Comment:
What's going on here?

Reply:
The previous behavior would skip nested compile/disable decorators.